framework_crates/bones_schema/src/alloc/vec.rs

```diff
@@ -337,6 +337,54 @@ impl SchemaVec {
         let b = &*(b as *const Self);
         a.eq(b)
     }
+
+    /// Remove and return the element at position `index` within the vector,
+    /// shifting all elements after it to the left.
+    /// # Panics
+    /// Panics if `index` is out of bounds.
+    pub fn remove(&mut self, index: usize) -> SchemaBox {
+        if index >= self.len {
+            panic!("index out of bounds");
+        }
+        let item = unsafe {
+            let ptr = self.buffer.unchecked_idx(index);
+            let mut boxed = SchemaBox::uninitialized(self.schema);
+            boxed
+                .as_mut()
+                .as_ptr()
+                .copy_from_nonoverlapping(ptr, self.schema.layout().size());
+
+            // Shift elements
+            let to_move = self.len - index - 1;
+            if to_move > 0 {
+                std::ptr::copy(
+                    self.buffer.unchecked_idx(index + 1),
+                    self.buffer.unchecked_idx(index),
+                    to_move * self.schema.layout().size(),
+                );
+            }
+
+            self.len -= 1;
+            boxed
+        };
+        item
+    }
+
+    /// Clears the vector, removing all values.
+    pub fn clear(&mut self) {
+        while self.pop_box().is_some() {}
+    }
+
+    /// Shortens the vector, keeping the first `len` elements and dropping the rest.
+    ///
+    /// If `len` is greater than the vector's current length, this has no effect.
+    pub fn truncate(&mut self, len: usize) {
+        if len < self.len {
```

Comment:
This `if` shouldn't be necessary because the while will just not run even once if `len < self.len`, right?